src/NServiceBus.Core/Serialization/SerializationConfigExtensions.cs

```diff
@@ -67,14 +67,31 @@ public static SerializationExtensions<T> AddDeserializer<T>(this EndpointConfigu
 
             var settings = new SettingsHolder();
             additionalSerializers.Add(Tuple.Create<SerializationDefinition, SettingsHolder>(serializationDefinition, settings));
-            return CreateSerializationExtension<T>(settings);
+            return CreateSerializationExtension<T>(settings, config.Settings);
         }
 
-        static SerializationExtensions<T> CreateSerializationExtension<T>(SettingsHolder settings) where T : SerializationDefinition
+        /// <summary>
+        /// Disables inference of message type based on the content type if the message type can't be determined by the 'NServiceBus.EnclosedMessageTypes' header.
+        /// </summary>
+        public static void DisableMessageTypeInference<T>(this SerializationExtensions<T> config) where T : SerializationDefinition
+        {
+            Guard.AgainstNull(nameof(config), config);
+
+            config.EndpointConfigurationSettings.Set(DisableMessageTypeInferenceKey, true);
```

Comment:
Wouldn't Settings work?
```suggestion
            config.Settings.Set(DisableMessageTypeInferenceKey, true);
```